pkg/tsdb/azuremonitor/metrics/url-builder.go

```diff
@@ -28,19 +29,21 @@ func (params *urlBuilder) buildResourceURI() *string {
 		subscription = params.DefaultSubscription
 	}
 
+	metricNamespace := params.MetricNamespace
+
 	if params.MetricNamespace == nil || *params.MetricNamespace == "" {
-		return nil
+		metricNamespace = params.MetricDefinition
```

Comment:
May be worth checking if this is `nil` or empty and if so throw an error, wdyt?

Reply:
Is there a case when a query would not have either a `MetricNamespace` or a `MetricDefinition`? I think the error is a good idea - I'm just genuinely curious 😃

Reply:
True - as discussed we want to prevent confusion if an alert is manually made. I added the error and the corresponding test.

Reply:
You'll probably need to update this to something like:
```go
	metricNamespace := params.MetricNamespace

	if metricNamespace == nil || *metricNamespace == "" {
		if params.MetricDefinition == nil || * params.MetricDefinition == "" {
			return nil, fmt.Errorf("no metricNamespace or metricDefiniton value provided")
		}
		metricNamespace = params.MetricDefinition
	}
```